### PR TITLE
Remove `[block]` intermediary

### PR DESCRIPTION
**Pull request by tienifr**
### Details

This PR completely removes `[block]` intermediary as it may coincide with user's input.

### Fixed Issues

$ https://github.com/Expensify/App/issues/17776
$ PROPOSAL: https://github.com/Expensify/App/issues/17776#issuecomment-1538347039

### Tests

1. Login ND with any account
2. Go to any chat
3. Type the word `[block]` and send it
4. Click the pencil icon to edit the comment
5. Verify that the composer shows `[block]` as is

- [x] Verify that no errors appear in the JS console

### Offline tests

NA

### QA Steps

1. Login ND with any account
2. Go to any chat
3. Type the word `[block]` and send it
4. Click the pencil icon to edit the comment
5. Verify that the composer shows `[block]` as is

- [ ] Verify that no errors appear in the JS console

### Screenshots/Videos
<details>
<summary>Web</summary>

[Screencast from 23-05-2023 21:12:02.webm](https://github.com/Expensify/expensify-common/assets/113963320/91b2d28e-43a8-4e63-aa7e-c092ab0ef4f4)

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/expensify-common/assets/113963320/786a05f4-71e6-425a-b11f-78a2d115864c

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/expensify-common/assets/113963320/74b54984-95d5-40d6-a456-1193a6562e37

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/expensify-common/assets/113963320/b3a56179-a121-430e-afe8-6ec8de9b4418

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/expensify-common/assets/113963320/d90846d0-5d7f-429f-a00d-1add25033a4a

</details>

<details>
<summary>Android</summary>

[untitled.webm](https://github.com/Expensify/expensify-common/assets/113963320/b2b950e5-52c7-404c-b095-490e96f35a5e)

</details>

### CODE REVIEW

**Comment by PauloGasparSv** [comment]
Assigning @cead22 and @rushatgabhane from the original issue here!

**Review by cead22 — CHANGES_REQUESTED** [review]
Just one small change. We should test that every piece of markdown we support is working correctly, but maybe those tests can go in the App PR, and we can get a full contributor checklist going in there as well

**Comment by cead22** [comment]
It's only a suggestion. We don't have to merge this PR in order to put the tests in the App PR, we can update the commit hash of expensify-common in package.json in app and test there, and once we're happy with the tests, we can merge this PR and update the commit hash again in the App PR.

The thing is we don't have the reviewer checklist here, but as long as we test sending and editing every type of markdown we support as part of the tests, that works for me

**Comment by tienifr** [comment]
Thanks! I will create another PR for regression testing. It'll be up really soon.

**Comment by tienifr** [comment]
The App PR is up here: https://github.com/Expensify/App/pull/19511.

File: lib/ExpensiMark.js [hunk 1]
```diff
@@ -511,17 +511,18 @@ export default class ExpensiMark {
...
+     * 1. We have text within the element.
      * 2. The text does not end with a new line.
      * 3. The text does not have quote mark '>' .
-     * 4. [block] not the last [block] in the string.
+     * 4. It's not the last element in the string.
      *
      * @param {String} htmlString
      * @returns {String}
      */
     replaceBlockWithNewLine(htmlString) {
```
[the remark this answers is not in the document]
Comment (tienifr) [reply lib/ExpensiMark.js:523]: `replaceBlockElementWithNewLine` sounds better to me.

**Comment by cead22** [comment]
>replaceBlockElementWithNewLine sounds better to me.

👍

**Review by cead22 — DISMISSED** [review]

**Comment by cead22** [comment]
@tienifr can you rebase the merge commit so it's signed so we can merge without changing the branch protection restrictions?

**Review by cead22 — APPROVED** [review]

**Pull request merged**